sphinxext/rediraffe.py

```diff
@@ -206,18 +214,22 @@ def build_redirects(app: Sphinx, exception: Union[Exception, None]) -> None:
         src_redirect_from = Path(PureWindowsPath(src_redirect_from))
         src_redirect_to = Path(PureWindowsPath(src_redirect_to))
 
-        # relative paths from source dir (without ext)
-        redirect_from = src_redirect_from.with_suffix("")
-        redirect_to = src_redirect_to.with_suffix("")
+        # remove extensions
+        redirect_from_name = remove_suffix(
+            src_redirect_from.name, app.config.source_suffix
+        )
+        redirect_to_name = remove_suffix(src_redirect_to.name, app.config.source_suffix)
+
+        redirect_from = src_redirect_from.parent / f"{redirect_from_name}.html"
+        redirect_to = src_redirect_to.parent / f"{redirect_to_name}.html"
 
         if type(app.builder) == DirectoryHTMLBuilder:
-            if redirect_from.name != "index":
-                redirect_from = redirect_from / "index"
-            if redirect_to.name != "index":
-                redirect_to = redirect_to / "index"
-
-        redirect_from = redirect_from.with_suffix(".html")
-        redirect_to = redirect_to.with_suffix(".html")
```

Comment:
```py        
		redirect_from = redirect_from.with_suffix(redirect_from.suffix + ".html")
        redirect_to = redirect_to.with_suffix(redirect_to.suffix + ".html")
```

I would prefer this as a fix as it doesn't require an extra custom path function.

Reply:
Yeh but I'd say it's incorrect to just remove any suffix; it should only be ones that map to suffixes of sphinx source files, which in principle can also contain dots like `.myst.md`.
So, in the context of sphinx, using `.with_suffix` and `.suffix` is not a good idea

Reply:
the code I added is exactly how sphinx does the suffix removal

Reply:
for example here 😉 https://github.com/sphinx-doc/sphinx/blob/9f44ee4dd1761e12a25b63671dcc4b8ace1f2555/sphinx/directives/other.py#L120

Reply:
do you still want me to change it?

Reply:
Ah, I hadn't considered multiple dots. This is fine then.